pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,385 @@
+		Long:  fmt.Sprintf(manageNodeLongDesc, fullName),
+		Run: func(c *cobra.Command, args []string) {
+			err := cfg.RunManageNode(f, out, c, args)
+			kcmdutil.CheckErr(err)
+		},
+	}
+	flags := cmd.Flags()
+
+	flags.BoolVar(&cfg.Schedulable, "schedulable", true, "Control pod schedulability on the node.")
+	flags.BoolVar(&cfg.Evacuate, "evacuate", true, "Migrate all/selected pods on the node.")
+	flags.BoolVar(&cfg.ListPods, "list-pods", true, "List all/selected pods on the node.")
+
+	flags.StringVar(&cfg.PodSelector, "pod-selector", "", "Label selector to filter pods on the node. Optional parameters for --evacuate or --list-pods")
+	flags.BoolVar(&cfg.DryRun, "dry-run", false, "Don't perform evacuate but show pods that will be migrated. Optional param for --evacuate")
+	flags.StringVar(&cfg.Selector, "selector", "", "Label selector to filter nodes. Either pass one/more nodes as arguments or use this node selector")
+
+	return cmd
+}
+
+// RunManageNode contains all the necessary functionality for the openshift cli manage-node command
+func (cfg *NodeConfig) RunManageNode(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	errList := []error{}
+	if errList = cfg.ValidateParams(cmd, args); len(errList) != 0 {
+		err := fmt.Errorf("(%v)", errList)
+		return kcmdutil.UsageError(cmd, err.Error())
+	}
+	if err := cfg.Configure(f, out); err != nil {
+		return err
+	}
+
+	cmdNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	nameArgs := []string{"nodes"}
+	if len(args) != 0 {
+		nameArgs = append(nameArgs, args...)
+	}
+
+	mapper, typer := f.Object()
+	r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).
+		SelectorParam(cfg.Selector).
+		ResourceTypeOrNameArgs(true, nameArgs...).
+		Flatten().
+		Do()
+	if r.Err() != nil {
+		return r.Err()
+	}
+
+	nodeCount := 0
+	ignoreHeaders := false
+	err = r.Visit(func(info *resource.Info) error {
+		nodeCount++
+		err = nil
+		node, ok := info.Object.(*kapi.Node)
+		if !ok {
+			err = fmt.Errorf("cannot convert input to Node: ", reflect.TypeOf(info.Object))
+			errList = append(errList, err)
+			return nil
+		}
+
+		if cmd.Flag("schedulable").Changed {
+			err = cfg.RunSchedulable(node, ignoreHeaders)
+		} else if cmd.Flag("evacuate").Changed && cfg.Evacuate {
+			if cfg.DryRun {
+				err = cfg.RunListPods(node)
+			} else {
+				err = cfg.RunEvacuate(node)
+			}
+		} else if cmd.Flag("list-pods").Changed && cfg.ListPods {
+			err = cfg.RunListPods(node)
+		}
+		ignoreHeaders = true
+
+		if err != nil {
+			errList = append(errList, err)
+			return nil
+		}
+		return nil
+	})
+	if len(errList) != 0 {
+		return fmt.Errorf("(%v)", errList)
+	}
+	if nodeCount == 0 {
+		return fmt.Errorf("No nodes found")
+	}
+	return nil
+}
+
+func NewNodeConfig() *NodeConfig {
+	return &NodeConfig{}
+}
+
+func (cfg *NodeConfig) ValidateParams(c *cobra.Command, args []string) []error {
```

Comment:
For ease of future delegation, `Validate` shouldn't need parameters.  Everything would set in the `NodeConfig` (or similar options object) and you could simply run the command.